src/status_im/chat/core.cljs

```diff
@@ -1,20 +1,23 @@
-(ns status-im.chat.core)
+(ns status-im.chat.core
+  (:require [status-im.data-store.messages :as messages-store]))
 
 ;; Seen messages
 (defn receive-seen
-  [chat-id sender {:keys [message-ids js-obj]} {:keys [db]}]
-  (merge
-   {:confirm-message-processed [{:web3   (:web3 db)
-                                 :js-obj js-obj}]}
-   (when-let [seen-messages-ids (-> (get-in db [:chats chat-id :messages])
-                                    (select-keys message-ids)
-                                    keys)]
-     {:db (reduce
-           (fn [new-db message-id]
-             (assoc-in new-db
-                       [:chats chat-id
-                        :messages message-id
-                        :user-statuses sender]
-                       :seen))
-           db
-           seen-messages-ids)})))
+  [chat-id sender {:keys [message-ids]} {:keys [db]}]
+  (when-let [seen-messages-ids (-> (get-in db [:chats chat-id :messages])
+                                   (select-keys message-ids)
+                                   keys)]
+    (let [new-db (reduce
+                  (fn [new-db message-id]
+                    (assoc-in new-db
+                              [:chats chat-id
+                               :messages message-id
+                               :user-statuses sender]
+                              :seen))
+                  db
+                  seen-messages-ids)]
+      {:db            new-db
+       :data-store/tx [(messages-store/update-messages-tx
```

Comment:
I take this is something we were not doing before, and this fixes it, correct?

Reply:
yes, we didn't update message's status before, which was a bug